### PR TITLE
Add Options to all connect and listen fns

### LINKED ISSUE #65: There should be one connect fn and one listen fn
- Move `SocketAddr` into `ClientConfig` / `ServerConfig`.
- Add a field for the remote or handle

### PR DESCRIPTION
**Pull request by tikue**
Fixes #65

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -3,40 +3,79 @@
...
 use serde::{Deserialize, Serialize};
 use std::io;
 use std::net::SocketAddr;
 use tokio_core::net::TcpListener;
-use tokio_core::reactor::Handle;
+use tokio_core::reactor::{self, Handle};
 use tokio_proto::BindServer;
 use tokio_service::NewService;
 
+/// Additional options to configure how the server starts up.
```
Comment (shaladdle) [inline src/server.rs]: nit: s/server starts up/how the server operates?

File: src/server.rs [hunk 2]
```diff
@@ -76,14 +115,16 @@ fn listener(addr: &SocketAddr, handle: &Handle) -> io::Result<TcpListener> {
...
 pub struct ListenFuture {
-    inner: futures::Oneshot<io::Result<SocketAddr>>,
+    inner: future::Either<futures::Oneshot<io::Result<SocketAddr>>,
+                          future::FutureResult<io::Result<SocketAddr>, futures::Canceled>>,
 }
 
 impl ListenFuture {
     #[doc(hidden)]
     pub fn from_oneshot(rx: futures::Oneshot<io::Result<SocketAddr>>) -> Self {
-        ListenFuture { inner: rx }
```
Comment (shaladdle) [inline src/server.rs]: It looks like this is never used now. Remove it? (or use it)

File: src/server.rs (hunk 1 above)
Comment (tikue) [inline src/server.rs]: Done.

File: src/server.rs (hunk 2 above)
Comment (tikue) [inline src/server.rs]: Done.

**Pull request merged**